backend/src/API/teamEventsAPI.ts

```diff
@@ -8,53 +8,54 @@ const teamEventAttendanceDao = new TeamEventAttendanceDao();
 export const getAllTeamEvents = async (user: IdolMember): Promise<TeamEvent[]> => {
   const canEditTeamEvents = await PermissionsManager.canEditTeamEvent(user);
   if (!canEditTeamEvents) throw new PermissionError('does not have permissions');
-  return TeamEventsDao.getAllTeamEvents();
+  const teamEvents = await TeamEventsDao.getAllTeamEvents();
+  return Promise.all(
+    teamEvents.map(async (event) => ({
+      ...event,
+      attendees: await teamEventAttendanceDao.getTeamEventAttendanceByEventId(event.uuid),
+      requests: []
+    }))
+  );
 };
 
 export const getAllTeamEventInfo = async (): Promise<TeamEventInfo[]> =>
   TeamEventsDao.getAllTeamEventInfo();
 
 export const createTeamEvent = async (
-  teamEvent: TeamEvent,
+  teamEventInfo: TeamEventInfo,
   user: IdolMember
-): Promise<TeamEvent> => {
+): Promise<TeamEventInfo> => {
   const canCreateTeamEvent = await PermissionsManager.canEditTeamEvent(user);
   if (!canCreateTeamEvent)
     throw new PermissionError('does not have permissions to create team event');
-  await TeamEventsDao.createTeamEvent(teamEvent);
-  return teamEvent;
+  await TeamEventsDao.createTeamEvent(teamEventInfo);
+  return teamEventInfo;
 };
 
 export const deleteTeamEvent = async (teamEvent: TeamEvent, user: IdolMember): Promise<void> => {
   if (!PermissionsManager.canEditTeamEvent(user)) {
     throw new PermissionError("You don't have permission to delete a team event!");
   }
+  await teamEventAttendanceDao.deleteTeamEventAttendance(teamEvent.uuid);
   await TeamEventsDao.deleteTeamEvent(teamEvent);
 };
 
 export const updateTeamEvent = async (
-  teamEvent: TeamEvent,
+  teamEventInfo: TeamEventInfo,
   user: IdolMember
-): Promise<TeamEvent> => {
+): Promise<TeamEventInfo> => {
   if (!PermissionsManager.canEditTeamEvent(user)) {
     throw new PermissionError(
       `User with email ${user.email} does not have permissions to update team events`
     );
   }
-  await TeamEventsDao.updateTeamEvent(teamEvent);
-  return teamEvent;
+  const updatedTeamEvent = await TeamEventsDao.updateTeamEvent(teamEventInfo);
+  return updatedTeamEvent;
 };
 
-export const requestTeamEventCredit = async (
-  uuid: string,
-  request: TeamEventAttendance
-): Promise<void> => {
-  const teamEvent = await TeamEventsDao.getTeamEvent(uuid);
-  const updatedTeamEvent = {
-    ...teamEvent,
-    requests: [...teamEvent.requests, request]
-  };
-  await TeamEventsDao.updateTeamEvent(updatedTeamEvent);
+export const requestTeamEventCredit = async (request: TeamEventAttendance): Promise<void> => {
+  const updatedteamEvent = { ...request, pending: true };
+  await teamEventAttendanceDao.createTeamEventAttendance(updatedteamEvent);
```

Comment:
Maybe we verify that the email of the member in the `TeamEventAttendance` parameter is the same as the email of the calling user?